pkg/io/video/scaler_cgo.go

```diff
@@ -92,3 +110,74 @@ func (f *FastNearestNeighbor) Scale(dst draw.Image, dr image.Rectangle, src imag
 		panic("unimplemented")
 	}
 }
+
+// FastBoxSampling is a CGO version of Box sampling scaler.
+type FastBoxSampling struct {
+}
+
+// Scale implements the draw.Scaler interface.
+func (f *FastBoxSampling) Scale(dst draw.Image, dr image.Rectangle, src image.Image, sr image.Rectangle, op draw.Op, opts *draw.Options) {
+	ddx, ddy := dr.Dx(), dr.Dy()
+	sdx, sdy := sr.Dx(), sr.Dy()
+	if sdx < ddx && sdy < ddy {
```

Comment:
`if sr.Dx() < dr.Dx() && sr.Dy() < dr.Dy() {` would be less confusing. Overhead is enough small.